Realm/RLMRealm.mm

```diff
@@ -382,6 +402,17 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
                 }
 
                 RLMRealmSetSchemaAndAlign(realm, schema);
+            } catch (SchemaMismatchException const& exception) {
+                if (configuration.deleteRealmIfMigrationNeeded) {
+                    BOOL success = [[NSFileManager defaultManager] removeItemAtURL:configuration.fileURL error:nil];
+                    if (success) {
+                        realm = nil;
+                        return [self realmWithConfiguration:configuration error:error];
```

Comment:
`invalidate` wouldn't do anything relevant. I think it shouldn't be needed with what we currently do before hitting this point, but calling `realm->_realm->close()` before setting `realm` to `nil` wouldn't hurt and would make things a bit more robust for future changes.